### ISSUE TITLE
Lock lines aren't locked

### ISSUE THREAD

**Issue by jheeffer** [vizabi/popbyage#33]
https://getsatisfaction.com/gapminder/topics/lock-lines-moving-on-demographic-pyramide

Below two screenshot, both locked at 2018, but different lock pop distribution shape.

Should be the same. At first I thought it was just scale jumps, but it's not just the scale (see screenshots), proportionality of bars also change.

![image](https://user-images.githubusercontent.com/1596071/48343704-5f578980-e673-11e8-97e3-ab3e009ded1e.png)
https://www.gapminder.org/tools/#$state$time$value=2018&step:10&offset:8;&entities$show$geo$/$in@=dnk;;;;&entities_age$grouping:10;&marker$color$which=g77_and_oecd_countries&palette$;;;;&ui$chart$lockNonSelected=2018;;&chart-type=popbyage

![image](https://user-images.githubusercontent.com/1596071/48343668-46e76f00-e673-11e8-9cac-1d45ea850b06.png)
https://www.gapminder.org/tools/#$state$time$value=2015&step:10&offset:5;&entities$show$geo$/$in@=dnk;;;;&entities_age$grouping:10;&marker$color$which=g77_and_oecd_countries&palette$;;;;&ui$chart$lockNonSelected=2018;;&chart-type=popbyage

**Comment by dab2000** [comment]
it's expected behavior imho.
we have 'age' grouping is 10 and current year is '2018' so we have frames ...1998,2008,2018,2028... with grouped real values by ages(0,10,20,30...). And we can 'lock' 2018 with values from '2018' data frame.

after  we move and release handle to 2015. we have 'age' grouping is 10 and current year is '2015' so we have frames ...1995,2005,2015,2025... with grouped real values by ages(0,10,20,30...). So we have 'lock' on  '2018' we rebuild lock line on interpolated data between 2015 and 2025. So lock line jump a little.

in addition:
if we move and release handle to 1998,2008,2028 etc. then lock line don't jump.
lock line don't jump in any case if 'age' grouping is 1.

**Comment by jheeffer** [comment]
I wouldn't say it's expected. It's expected that the data of 2018 always stays the same. It might be expected from the current implementation, but it's not expected from a user's and data standpoint.

So, to do it right, the 2018 lock frame in 2015 shouldn't be an interpolation between 2015 and 2025 but the actual data of 2018.
Is that possible?

**Comment by jheeffer** [comment]
Furthermore, I'm not sure why there's any interpolation at all for yearly data if we have actual yearly data in the dataset?

**Comment by dab2000** [comment]
we have 'step' prop in 'time' model and build frames for timesteps only

**Comment by jheeffer** [comment]
So, why are popbyage group size and time step related (I see they are always equal)? I don't see why that is necessary? I should be able to have steps of 1 year with 10 year age groups, right?

**Comment by dab2000** [comment]
it's not problem use 2018 data for current session but it can be some harder to get 2018 data on reload
(set time value to 2018 for get values for lock then return value to 2015). i need to check

**Comment by dab2000** [comment]
group size and time step related for proper pyramid animation

**Comment by jheeffer** [comment]
please elaborate : )
I can imagine a proper animation without related group size and time step I think. 
The only problem might be stopping on a year where the bottom group is incomplete (e.g. 5 year groups, lower is 2 years). But in that case we can reset the offset to 0, like we do when dragging?

**Comment by dab2000** [comment]
yes animation can be done with step=1 and any group size. but we need move all snap and step logic to tool component.

**Comment by jheeffer** [comment]
Why?

**Comment by dab2000** [comment]
for group size 10 the animation stops should be every 10 years
